### PR TITLE
Add nanorpc trait

### PR DESCRIPTION
**Pull request by antidmg**
Use a `nanorpc` macro to derive a client/server for existing `NodeServer` methods.

### CODE REVIEW

**Comment by antidmg** [comment]
@nullchinchilla @sshmendez I'm in the process of implementing this inside `themelio-node`, but wanted to get a high-level review of this trait first to make sure it looks alright.

**Review by nullchinchilla — COMMENTED** [review]
Yeah I think generally it looks good, except for the fact that get_summary doesn't need to return an option (because "there isn't any network summary to return" isn't a possible condition). Option here really encodes the possibility of absence, which is possible when the block height is out of range :)

**Comment by antidmg** [comment]
I'm using `melnet2` in `[dev-dependencies]`, so CI will complain until that's published as a crate. (I can't find it in a search on crates.io yet)

File: examples/nodeprot-cli.rs [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        let rpc_client = NodeRpcClient(
+            backhaul
+                .connect(args.addr.to_string().into())
+                .await
+                .expect("failed to create RPC client"),
+        );
+
+        // one-off set up to test using a custom network.
+        let client = ValClient::new(args.netid, rpc_client);
+        let snapshot = client.insecure_latest_snapshot().await.unwrap();
```
Comment (antidmg) [inline examples/nodeprot-cli.rs:25]: I had to add this method back, since it was the only way I could find lets me trust a custom network.

**Pull request merged**